UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs

```diff
@@ -1221,22 +1237,75 @@ private void AddLayer<T>(LayerType type, string name) where T : Layer.LayerData,
                 }
             }
 
+            int layerDepth = 0;
+            if (room.Layers.Count > 0)
+            {
+                layerDepth = room.Layers.Select(l => l.LayerDepth).Max();
+                if (layerDepth + 100 > int.MaxValue)
+                {
+                    if (layerDepth + 1 > int.MaxValue)
+                    {
+                        layerDepth -= 1;
+                        MainWindow.ShowWarning("Warning - a maximum layer depth is reached.\nYou probably should change the depth of the new layer.");
+                    }
+                    else
+                        layerDepth += 1;
+                }
+                else
+                    layerDepth += 100;
+            }
+
+            string baseName = null;
+            int nameNum = 0;
+            while (room.Layers.Any(l => l.LayerName.Content == name))
+            {
+                if (baseName is null)
+                {
+                    // "name123" => "123"
+                    Match numMatch = endNumberEx.Match(name);
+
+                    if (numMatch.Success)
+                    {
+                        if (baseName is null)
+                        {
+                            baseName = name[..^numMatch.Length];
+                            nameNum = Int32.Parse(numMatch.Groups[0].Value) + 1;
+                        }
+                        else
+                            nameNum++;
+
+                        // "name9" => "name10" 
+                        name = baseName + nameNum;
+                    }
+                    else
+                    {
+                        // "name" => "name1"
+                        name += 1;
+                    }
+                }
+                else
+                    name = baseName + (++nameNum);
+            }
+
             Layer layer = new();
             layer.LayerName = data.Strings.MakeString(name);
             layer.LayerId = largest_layerid + 1;
             layer.LayerType = type;
+            layer.LayerDepth = layerDepth;
             layer.Data = new T();
             room.Layers.Add(layer);
             room.UpdateBGColorLayer();
 
             if (layer.LayerType == LayerType.Assets)
             {
-                // create a new pointer list (if null)
+                // "??=" - assign if null
                 layer.AssetsData.LegacyTiles ??= new UndertalePointerList<Tile>();
-                // create new sprite pointer list (if null)
                 layer.AssetsData.Sprites ??= new UndertalePointerList<SpriteInstance>();
-                // create new sequence pointer list (if null)
                 layer.AssetsData.Sequences ??= new UndertalePointerList<SequenceInstance>();
+
+                // if GMS version is less that 2.3.2
+                if (!data.GMS2_3_2)
```

Comment:
Comment is incorrect or check is incorrect.
`!data.GM2_3_2` can also get triggered on versions above gms232

Reply:
That's weird, I'm pretty sure they're still used in 2022 something right now.

Reply:
No, I think only the source comment and my comment should be rewritten